api/routes/apds/activities/expenses/put.js

```diff
@@ -4,103 +4,92 @@ const {
   apdActivityExpense: defaultActivityExpenseModel,
   apdActivityExpenseEntry: defaultActivityExpenseEntryModel
 } = require('../../../../db').models;
-const { userCanEditAPD: defaultUserCanEditAPD } = require('../../utils');
-const loggedIn = require('../../../../auth/middleware').loggedIn;
+const {
+  loggedIn,
+  loadActivity,
+  userCanEditAPD
+} = require('../../../../middleware');
 
 module.exports = (
   app,
   ActivityModel = defaultActivityModel,
   ExpenseModel = defaultActivityExpenseModel,
-  ExpenseEntryModal = defaultActivityExpenseEntryModel,
-  userCanEditAPD = defaultUserCanEditAPD
+  ExpenseEntryModal = defaultActivityExpenseEntryModel
 ) => {
   logger.silly('setting up PUT /activities/:id/expenses route');
-  app.put('/activities/:id/expenses', loggedIn, async (req, res) => {
-    logger.silly(req, 'handling PUT /activities/:id/expenses route');
-    logger.silly(
-      req,
-      `attempting to update expenses on activity [${req.params.id}]`
-    );
-
-    try {
-      const activityID = +req.params.id;
-      const activity = await ActivityModel.where({ id: activityID }).fetch({
-        withRelated: ['apd', 'expenses.entries']
-      });
+  app.put(
+    '/activities/:id/expenses',
+    loggedIn,
+    loadActivity(),
+    userCanEditAPD(ActivityModel),
+    async (req, res) => {
+      logger.silly(req, 'handling PUT /activities/:id/expenses route');
+      logger.silly(
+        req,
+        `attempting to update expenses on activity [${req.params.id}]`
+      );
 
-      if (!activity) {
-        logger.verbose(req, 'activity not found');
-        return res.status(404).end();
-      }
-
-      const apdID = activity.related('apd').get('id');
-      if (!await userCanEditAPD(req.user.id, apdID)) {
-        logger.verbose(
-          req,
-          'user (state) not associated with the apd this activity belongs to'
-        );
-        return res.status(404).end();
-      }
```

Comment:
That new middleware means these two checks don't need to be anywhere else anymore.  I think we had them in 3-4 places.

Reply:
sweeeet